sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/LiteralExpressionProtoConverter.scala

```diff
@@ -110,15 +110,17 @@ object LiteralExpressionProtoConverter {
           proto.DataType.newBuilder
             .setArray(LiteralValueProtoConverter.getProtoArrayType(lit.getArray))
             .build())
-        expressions.Literal.create(arrayData, dataType)
+        val convert = CatalystTypeConverters.createToCatalystConverter(dataType)
+        expressions.Literal(convert(arrayData), dataType)
```

Comment:
`LiteralValueProtoConverter.toCatalystArray` does not covert everything to the catalyst format?

Reply:
@cloud-fan No, only partially. Maps and structs also have this issue. There are a few problems we need to fix so that LiteralExpressionProtoConverter can convert everything to the Catalyst format. I’ve fixed some of them in this PR.